src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -399,18 +405,48 @@ private function buildConstructor(ClassBuilder $class): void
 ', ['PROPERTY' => $p->getName(), 'ORG_NAME' => $p->getOriginalName()]);
         }
 
-        $body .= '
+        if ($class->shouldAllowExtraKeys()) {
+            $body .= '
+    $this->_extraKeys = $value;
+';
+        } else {
+            $body .= '
     if ([] !== $value) {
         throw new InvalidConfigurationException(sprintf(\'The following keys are not supported by "%s": \', __CLASS__).implode(\', \', array_keys($value)));
     }';
 
-        $class->addUse(InvalidConfigurationException::class);
+            $class->addUse(InvalidConfigurationException::class);
+        }
+
         $class->addMethod('__construct', '
 public function __construct(array $value = [])
 {
 '.$body.'
-}
-');
+}');
+    }
+
+    private function buildSetExtraKey(ClassBuilder $class): void
+    {
+        if (!$class->shouldAllowExtraKeys()) {
+            return;
+        }
+
+        $class->addProperty('_extraKeys');
+
+        $class->addMethod('set'.substr($class->getName(), 0, -6).'ExtraKey', '
```

Comment:
For SwiftBundle, this would be `setStreamOptionsExtraKey()`, but other setters do not have the `set` prefix. 

```php
use Symfony\Config\SwiftmailerConfig;

return static function (SwiftmailerConfig $swift) {
     $swift->mailer('main')
         ->streamOptions()->setStreamOptionsExtraKey('foo', 'bar');
};
```

Could we make it more "natural"?

```php
use Symfony\Config\SwiftmailerConfig;

return static function (SwiftmailerConfig $swift) {
     $swift->mailer('main')
         ->streamOptions()->set('foo', 'bar');
};
```

Reply:
@Nyholm Yep, `set` looks better, done.